### PR TITLE
Update for `fileprivate`

### PR DESCRIPTION
**Pull request by CodaFi**
Implements the access control changes needed for the Swift Package Manager to compile under apple/swift#3000.

### CODE REVIEW

**Comment by ddunbar** [comment]
👏

File: Sources/Basic/FSProxy.swift [hunk 1]
```diff
@@ -268,19 +268,19 @@ private class LocalFS: FSProxy {
 //
 // FIXME: This class does not yet support concurrent mutation safely.
 public class PseudoFS: FSProxy {
-    private class Node {
+    fileprivate class Node {
```
Comment (ddunbar) [inline Sources/Basic/FSProxy.swift]: This surprises me, it is certainly the intent that this is `private` not `fileprivate`. What required this?

Comment (CodaFi) [inline Sources/Basic/FSProxy.swift]: I remember thinking that too.  Probably a bogus diagnostic at one point.  I'll change it

Comment (CodaFi) [inline Sources/Basic/FSProxy.swift]: Ah, I remember now.  The rule for `private` is the inner scope can refer to `private` members in the outer scope but not vice-versa.

Comment (ddunbar) [inline Sources/Basic/FSProxy.swift]: Huh, this seems unfortunate for exactly this use case. Declaring it as fileprivate *looks* wrong... is it possible this indicates a problem in the proposal?

Comment (CodaFi) [inline Sources/Basic/FSProxy.swift]: It does seem wrong to use `file` to mean `scope`, but it was the name the community settled on 😅

**Comment by anandabits** [comment]
private should work here exactly as you expect @ddunbar.  The proposal is vaguely worded and that is causing confusion.  @CodaFi started a thread on the mailing list https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160613/021141.html about this.  Everyone who participated in the original discussion and review has the same interpretation of the intent of the proposal as @ddunbar.  @DougGregor, as review manager can you please confirm the intended semantics of private so implementation can move forward.

**Comment by anandabits** [comment]
I would also like to add that the migration path provided by @CodaFi is the correct *automated* migration - `fileprivate` preserves the existing semantics precisely.  Changing this to `private` takes advantage of the *new* semantics introduced by SE-0025.  

It is probably reasonable to discuss whether this moves forward first as a semantic-preserving migration and then Swift PM does a review to tighten visibility, or whether that review should be part of the initial migration.

**Comment by ddunbar** [comment]
Thanks for the clarification @anandabits, that is a relief -- I do strongly prefer the interpretation you are espousing.

**Comment by CodaFi** [comment]
We need to not do this kind of clarification here.  Let's get this in writing on SE-0025 before any of this goes through.  This patch may need to be amended soon anyhow.

**Comment by ddunbar** [comment]
I was just referring to the reference to the thread... I can't keep up with all the lists so I appreciate pointers when useful. I agree official discussion/decisions needs to be on the list.

**Comment by ddunbar** [comment]
Closing per #476

**Pull request closed without merging**